docs/gallery/plot_aligneddynamictable.py

```diff
@@ -0,0 +1,323 @@
+#     * Interact with :py:class:`~hdmf.common.alignedtable.AlignedDynamicTable` as well as
+#       the category (sub-tables) it contains in the same fashion as with
+#       :py:class:`~hdmf.common.table.DynamicTable`
+#
+# When to use (and not use) AlignedDynamicTable?
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# :py:class:`~hdmf.common.alignedtable.AlignedDynamicTable` is a useful data structure but it is also
+# fairly complex, consisting of multiple :py:class:`~hdmf.common.table.DynamicTable` objects, each of
+# which is itself a complex type composed of many datasets and attributes. In general, if a simpler
+# data structure is sufficient, then consider using those instead. For example, consider using instead:
+#
+# * :py:class:`~hdmf.common.table.DynamicTable` if a regular table is sufficient.
+# * A compound dataset via :py:class:`~hdmf.container.Table` if all columns of a table are fixed
+#   and fast, column-based access is not critical but fast row-based access is.
+# * Multiple, separate tables if using :py:class:`~hdmf.common.alignedtable.AlignedDynamicTable` would
+#   lead to duplication of data (i.e., de-normalize data), e.g., by having to replicate values across
+#   rows of the table.
+#
+# Use :py:class:`~hdmf.common.alignedtable.AlignedDynamicTable` when:
+#
+# * When you need to group columns in a :py:class:`~hdmf.common.table.DynamicTable` by category
+# * Need to avoid name collisions between columns in a :py:class:`~hdmf.common.table.DynamicTable`
+#   and creating compound columns is not an option
+#
+
+###############################################################################
+# Constructing a table
+# --------------------
+#
+# To create an :py:class:`~hdmf.common.alignedtable.AlignedDynamicTable`, call the constructor with:
+#
+# * ``name`` string with the name of the table, and
+# * ``description`` string to describe the table.
+#
+
+# sphinx_gallery_thumbnail_path = 'figures/gallery_thumbnail_aligneddynamictable.png'
+from hdmf.common import AlignedDynamicTable
+
+customer_table = AlignedDynamicTable(
+    name='customers',
+    description='an example aligned table',
+)
+
+###############################################################################
+# Initializing columns of the primary table
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# The basic behavior of adding data and initalizing  :py:class:`~hdmf.common.alignedtable.AlignedDynamicTable`
+# is the same as in :py:class:`~hdmf.common.table.DynamicTable`. See the :doc:`DynamicTable tutorial  <dynamictable>`
+# for details. E.g., using  the ``columns`` and ``colnames`` parameters (which are inherited from
+# :py:class:`~hdmf.common.table.DynamicTable`) we can define the columns of the primary table.
+# All columns must have the same length.
+
+from hdmf.common import VectorData
+
+col1 = VectorData(
+    name='firstname',
+    description='Customer first name',
+    data=['Peter', 'Emma'])
+col2 = VectorData(
```

Comment:
Super minor: suggested style change
```suggestion
    data=['Peter', 'Emma']
)
col2 = VectorData(
```

Reply:
See also similar lines below...

Reply:
Fixed throughout